ZeeguuAPI/ZeeguuAPI/ZeeguuAPI.swift

```diff
@@ -0,0 +1,334 @@
+	public func translateWord(word: String, context: String, url: String, completion: (translation: String?) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(translation: nil)
+		}
+		
+		self.learnedAndNativeLanguage { (dict) -> Void in
+			if (dict != nil) {
+				if let learned = dict!["learned"].string, native = dict!["native"].string {
+					let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.Translate, pathComponents: [learned, native], method: HTTPMethod.POST, parameters: ["context": context, "word": word, "url": url])
+					self.sendAsynchronousRequest(request) { (response, error) -> Void in
+						if (response != nil) {
+							completion(translation: response!)
+						} else {
+							completion(translation: nil)
+						}
+					}
+				}
+			}
+			completion(translation: nil)
+		}
+	}
+	
+	/// Adds the translation of the given word to the user's bookmarks and returns its ID.
+	///
+	/// - parameter word: The word to bookmark.
+	/// - parameter translation: The translation of `word`.
+	/// - parameter context: The context in which the word appeared.
+	/// - parameter url: The url of the article in which the word was found.
+	/// - parameter title: The title of the article in which the word was found.
+	/// - parameter completion: A block that will receive a string containing the id of the newly made bookmark.
+	public func bookmarkWord(word: String, translation: String, context: String, url: String, title: String?, completion: (bookmarkID: String?) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(bookmarkID: nil)
+		}
+		
+		self.learnedAndNativeLanguage { (dict) -> Void in
+			if (dict != nil) {
+				if let learned = dict!["learned"].string, native = dict!["native"].string {
+					var params = ["context": context, "url": url]
+					if (title != nil) {
+						params["title"] = title
+					}
+					let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.BookmarkWithContext, pathComponents: [learned, word, native, translation], method: HTTPMethod.POST, parameters: params)
+					self.sendAsynchronousRequest(request) { (response, error) -> Void in
+						if (response != nil) {
+							completion(bookmarkID: response!)
+						} else {
+							completion(bookmarkID: nil)
+						}
+					}
+				}
+			}
+			completion(bookmarkID: nil)
+		}
+	}
+	
+	/// Deletes the bookmark with the given ID.
+	///
+	/// - parameter bookmarkID: The ID of the bookmark to delete.
+	/// - parameter completion: A block that will receive a boolean indicating if the bookmark could be deleted or not.
+	public func deleteBookmarkWithID(bookmarkID: String, completion: (success: Bool) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(success: false)
+		}
+		
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.DeleteBookmark, pathComponents: [bookmarkID], method: HTTPMethod.POST, parameters: nil)
```

Comment:
Nice. Looks cleaner! 
But that reference to `.AddUser` is puzzling. 
I guess I really need to read more about this language..

Where do you declare that you're importing the ZeeguuAPIEndpoint? Or can you use the .AddUser just because you specified the type in the method declaration?